app/src/test/java/org/simple/clinic/instantsearch/InstantSearchUpdateTest.kt

```diff
@@ -290,6 +301,77 @@ class InstantSearchUpdateTest {
+    val facility = TestData.facility(
+        uuid = UUID.fromString("885c6339-9a96-4c8d-bfea-7eea74de6862"),
+    )
+    val searchQueryModel = defaultModel
+        .facilityLoaded(facility)
+        .searchQueryChanged("Pat")
+
+    val indiaNationalHealthID = "28-3123-2283-6682"
+    val patientPrefillInfo = TestData.indiaNHIDInfoPayload(
+        healthIdNumber = indiaNationalHealthID
+    ).toPatientPrefillInfo()
+
+    val identifier = Identifier(indiaNationalHealthID, IndiaNationalHealthId)
+
+    val ongoingNewPatientEntry = OngoingNewPatientEntry(
+        personalDetails = OngoingNewPatientEntry.PersonalDetails(
+            fullName = patientPrefillInfo.fullName,
+            dateOfBirth = patientPrefillInfo.dateOfBirth.format(DateTimeFormatter.ofPattern("dd/MM/yyyy")),
+            gender = Gender.Male,
+            age = null),
+        address = OngoingNewPatientEntry.Address.BLANK.withColonyOrVillage(patientPrefillInfo.address),
+        identifier = identifier)
+
+    updateSpec
+        .given(searchQueryModel.patientPrefillInfoUpdated(patientPrefillInfo).additionalIdentifierUpdated(identifier))
+        .whenEvent(RegisterNewPatientClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(SaveNewOngoingPatientEntry(ongoingNewPatientEntry))
+        ))
+  }
+
+  @Test
+  fun `when register new patient is clicked and patient prefill info is not empty and dob is in wrong format, then save it in ongoing patient entry with dob in correct format`(){
+    val facility = TestData.facility()
+    val searchQueryModel = defaultModel
+        .facilityLoaded(facility)
+        .searchQueryChanged("Pat")
+
+    val dateOfBirthWithWrongFormat = LocalDate.parse("2012-03-02")
+    val dateOfBirthWithCorrectFormat = "02/03/2012"
+
+    val indiaNationalHealthID = "28-3123-2283-6682"
+    val patientPrefillInfo = TestData.indiaNHIDInfoPayload(
+        healthIdNumber = indiaNationalHealthID,
+        dateOfBirth = dateOfBirthWithWrongFormat
+    ).toPatientPrefillInfo()
+
+    val identifier = Identifier(indiaNationalHealthID, IndiaNationalHealthId)
+
+    // ongoing new patient entry with correct format
+    val ongoingNewPatientEntry = OngoingNewPatientEntry(
+        personalDetails = OngoingNewPatientEntry.PersonalDetails(
+            fullName = patientPrefillInfo.fullName,
+            dateOfBirth = dateOfBirthWithCorrectFormat,
+            gender = Gender.Male,
+            age = null),
+        address = OngoingNewPatientEntry.Address.BLANK.withColonyOrVillage(patientPrefillInfo.address),
+        identifier = identifier)
+
+    updateSpec
+        .given(searchQueryModel.patientPrefillInfoUpdated(patientPrefillInfo).additionalIdentifierUpdated(identifier))
+        .whenEvent(RegisterNewPatientClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(SaveNewOngoingPatientEntry(ongoingNewPatientEntry))
+        ))
+  }
```

Comment:
I think I may not have been clear earlier. What I mean is that this test is unnecessary since a `LocalDate` itself does not have anything to do with the format once it is parsed.

Since the actual parsing here is handled by the `ScanSimpleIdScreen`, in the event that the DOB in the scanned QR code is a different format than we expect, the parsing will fail there and we will never end up in this case since the `LocalDate` will never get created in the first place.

Hence, this test is unnecessary and we can remove it.

Does this make sense?